options.go

```diff
@@ -84,6 +87,12 @@ func WithCredentialsJSON(b []byte) Option {
 			d.err = errtype.NewConfigError(err.Error(), "n/a")
 			return
 		}
+		ud, err := c.GetUniverseDomain()
+		if err != nil {
+			d.err = errtype.NewConfigError(err.Error(), "n/a")
+			return
+		}
+		d.authUniverseDomain = ud
```

Comment:
Can we avoid setting this if it's the default universe?

Reply:
Probably but would that really simplify anything later in the flow? We would still need to set it if it were non default universe.